api/awsauth/dynamo_client.py

```diff
@@ -0,0 +1,53 @@
+from typing import Dict, List, Any, Optional
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+
+class DynamoDBClient:
+    def __init__(self, client=None):
+        self._client = client or boto3.resource("dynamodb")
+
+    def get_item(self, table: str, key: Dict[str, Any]) -> Optional[Dict[Any, Any]]:
+        """Get single item from table.
+
+        Args:
+            table: Table name
+            key: Key to query table with. Should be in form {'<attribute name>': <attribute value>}.
+
+        Returns: Row if exists, None if missing.
+        """
+        table = self._client.Table(table)
+        result = table.get_item(Key=key)
+
+        if "Item" not in result:
+            return None
+
+        return result["Item"]
+
+    def query_index(self, table: str, index: str, key: str, value: Any) -> List[dict]:
+        """Query index for items where `key` == `value`.
+
+        Args:
+            table: Table name.
+            index: Index name.
+            key: Key to query.
+            value: value.
+
+        Returns: List of rows matching query.
+        """
+        table = self._client.Table(table)
+
+        result = table.query(IndexName=index, KeyConditionExpression=Key(key).eq(value))
+        return result["Items"]
+
+    def put_item(self, table: str, item: Dict[str, Any]) -> None:
+        """Add item to table.
+
+        Args:
+            table: Table name.
+            item: Item to add in form {'<attribute_name>': <attribute_value>, ...}.
+
+        """
+        table = self._client.Table(table)
+        table.put_item(Item=item)
```

Comment:
How about something like ConditionExpression='attribute_not_exists(email)'  to avoid duplicates if two requests are racing to create a new item?